src/Featurizers/StringFeaturizer.h

```diff
@@ -88,9 +137,85 @@ class StringEstimator : public Components::InferenceOnlyEstimatorImpl<StringTran
 // |  StringTransformer
 // |
 // ----------------------------------------------------------------------
+template <typename T>
+StringTransformer<T>::StringTransformer(bool useEmptyStringsForNullValues /*=false*/) :
+    _useEmptyStringsForNullValues(
+        std::move(
+            [&useEmptyStringsForNullValues](void) -> bool & {
+                if(useEmptyStringsForNullValues && Traits<T>::IsNullableType == false)
+                    throw std::invalid_argument("empty strings cannot be used with types that are not nullable");
+
+                return useEmptyStringsForNullValues;
+            }()
+        )
+    ),
+    _impl(_useEmptyStringsForNullValues ? ToStringWithEmptyStringsForNullValues : ToString) {
+}
+
 template <typename T>
 StringTransformer<T>::StringTransformer(Archive &ar) :
-    BaseType(ar) {
+    StringTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            return StringTransformer(Traits<bool>::deserialize(ar));
+        }()
+    ) {
+}
+
+template <typename T>
+bool StringTransformer<T>::operator==(StringTransformer const &other) const {
+    return _useEmptyStringsForNullValues == other._useEmptyStringsForNullValues;
```

Comment:
I have never done an `==` with templated classes. Will it handle the type checking automatically? Or will it say that `StringTransformer<float> == StringTransformer<double>` if they both have `true` for `_useEmptyStringsForNullValues `?

Reply:
That would result in a compiler error (because the type `StringTransformer<float>` is not the same as the type `StringTransformer<double>`. If we wanted to implement cross-type equality, we could redefine this as `template <typename OtherT> bool operator==(StringTransformer<OtherT> const &other) const`.